presto-orc/src/main/java/com/facebook/presto/orc/OrcRecordReader.java

```diff
@@ -362,14 +365,19 @@ public int nextBatch()
             }
         }
 
-        currentBatchSize = toIntExact(min(maxBatchSize, currentGroupRowCount - nextRowInGroup));
-
+        // nextBatchSize is used to grow the batch size from INITIAL_BATCH_SIZE by the factor of BATCH_SIZE_GROWTH_FACTOR.
+        // Note that the maxBatchSize will be recalculated in readBlock() for each read, and cannot be greater than MAX_BATCH_SIZE
+        currentBatchSize = toIntExact(min(nextBatchSize, min(maxBatchSize, currentGroupRowCount - nextRowInGroup)));
         for (StreamReader column : streamReaders) {
             if (column != null) {
                 column.prepareNextRead(currentBatchSize);
             }
         }
         nextRowInGroup += currentBatchSize;
+
+        // Grow the next batch size by BATCH_SIZE_GROWTH_FACTOR. It will be used to calculate the
+        // currentBatchSize for the next batch
+        nextBatchSize = currentBatchSize * BATCH_SIZE_GROWTH_FACTOR;
```

Comment:
It's not needed because when we calculate the currentBatchSize for next batch, it will always be capped by maxBatchSize which was again capped by MAX_BATCH_SIZE.

Reply:
But if you think it'll help people understanding I'll add the check.